pkg/chart/loader/load.go

```diff
@@ -103,6 +104,9 @@ func LoadFiles(files []*BufferedFile) (*chart.Chart, error) {
 		// Deprecated: requirements.yaml is deprecated use Chart.yaml.
 		// We will handle it for you because we are nice people
 		case f.Name == "requirements.yaml":
+			if c.Metadata.APIVersion != chart.APIVersionV1 {
+				log.Printf("Warning: Dependencies are handled in Chart.yaml since apiVersion \"v2\". We recommend migrating dependencies to Chart.yaml.")
+			}
```

Comment:
This is a warning if the `apiVersion` is `v2` and there is a `requirements.yaml` file. Would it be too much to remove the `requirements.yaml` from the `v2` chart and make this an error?

Reply:
@mattfarina There was feed back from @bacongobbler in https://github.com/helm/helm/pull/6117#discussion_r308900043 about handling the requirements file and giving a warning.

Reply:
Because the translation from requirements.yaml to Chart.yaml are 1:1 there's no technical reason why we can't support both in apiVersion v2.

Reply:
@bacongobbler @adamreese @hickeyma Are we thinking of these as specs we can independently document and have multiple tools work on things that conform to the specs? With the `v1` we didn't do that even though there was a task for it. If we do think of it as a spec... than what does the spec say about the location of the requirements? If one of us writes the spec on this how do we describe the differences?

Reply:
I'll defer to @adamreese, but you do raise good points RE: third-party tools looking to support apiVersion v2.

Reply:
@mattfarina would it be fair to move forward with this PR as-is and carry on the discussion in a ticket? This retains the same functionality as before. We can discuss whether or not we want to not support requirements.yaml in apiVersion v2 as a separate PR. How does that sound?

Reply:
That sounds fine. I started #6151 to have that conversation.

Note, the whole reason I thought of a spec being written out is a) I've been asked about one and b) it would cause us to think some things through that we may be missing.

Reply:
@mattfarina Good idea. Putting it out in the open will help us and the users afterwards.